packages/lit-virtualizer/src/layouts/flow.ts

```diff
@@ -370,42 +370,43 @@ export class FlowLayout extends BaseLayout<BaseLayoutConfig> {
     items.set(this._anchorIdx, {pos: this._anchorPos, size: anchorSize});
 
     this._first = (this._last = this._anchorIdx);
-    this._physicalMin = this._anchorPos;
-    this._physicalMax = this._anchorPos + anchorSize;
+    this._physicalMin = this._anchorPos - anchorLeadingMargin;
+    this._physicalMax = this._anchorPos + anchorSize + anchorTrailingMargin;
 
     while (this._physicalMin > lower && this._first > 0) {
       let size = this._getSize(--this._first);
       if (size === undefined) {
         this._stable = false;
         size = this._getAverageSize();
       }
-      let margin = this._metricsCache.getMarginSize(this._first + 1);
+      let margin = this._metricsCache.getMarginSize(this._first);
       if (margin === undefined) {
         this._stable = false;
         margin = this._metricsCache.averageMarginSize;
       }
-      this._physicalMin -= size + margin;
+      this._physicalMin -= size;
       const pos = this._physicalMin;
       items.set(this._first, {pos, size});
+      this._physicalMin -= margin;
       if (this._stable === false && this._estimate === false) {
         break;
       }
     }
 
     while (this._physicalMax < upper && this._last < this._totalItems - 1) {
-      let margin = this._metricsCache.getMarginSize(++this._last);
-      if (margin === undefined) {
-        this._stable = false;
-        margin = this._metricsCache.averageMarginSize;
-      }
-      let size = this._getSize(this._last);
+      let size = this._getSize(++this._last);
       if (size === undefined) {
         this._stable = false;
         size = this._getAverageSize();
       }
-      const pos = this._physicalMax + margin;
+      let margin = this._metricsCache.getMarginSize(this._last);
+      if (margin === undefined) {
+        this._stable = false;
+        margin = this._metricsCache.averageMarginSize;
+      }
+      const pos = this._physicalMax;
       items.set(this._last, {pos, size});
-      this._physicalMax += margin + size;
+      this._physicalMax += size + margin;
       if (this._stable === false && this._estimate === false) {
```

Comment:
Nit alert - no action required. The type of `this._stable` and `this._estimate` is only boolean. Can this be simplified to: `(!this.stable && !this._estimate)`?

Reply:
Fixed!